### PR TITLE
S32K1XX: FlexCAN fixes & EEEPROM driver

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
## Summary
Fixes compile error with older compiler for FlexCAN & SocketCAN
FlexCAN don't disable interrupt but only remove masks TX MB interrupts
S32K1XX: Emulated EEPROM block device driver

## Impact
S32K1XX & Kinetis FlexCAN driver

## Testing
Verified on S32K1XX

### CODE REVIEW

File: arch/arm/src/kinetis/kinetis_flexcan.c [hunk 1]
```diff
@@ -719,7 +719,8 @@ static int kinetis_transmit(FAR struct kinetis_driver_s *priv)
 
       uint32_t *frame_data_word = (uint32_t *)&frame->data[0];
 
-      for (int i = 0; i < (frame->len + 4 - 1) / 4; i++)
+      uint32_t i;
```
Comment (acassis) [inline arch/arm/src/kinetis/kinetis_flexcan.c:722]: Although code inside arch/ are not required to follow C89, please move the function declaration to the top of the function to spread good practice and avoid people follow it to core parts of the OS where it is forbidden.

File: arch/arm/src/kinetis/kinetis_flexcan.c [hunk 2]
```diff
@@ -863,7 +864,8 @@ static void kinetis_receive(FAR struct kinetis_driver_s *priv,
 
           uint32_t *frame_data_word = (uint32_t *)&frame->data[0];
 
-          for (int i = 0; i < (frame->len + 4 - 1) / 4; i++)
+          uint32_t i;
```
Comment (acassis) [inline arch/arm/src/kinetis/kinetis_flexcan.c:867]: ditto

File: arch/arm/src/kinetis/kinetis_flexcan.c [hunk 3]
```diff
@@ -978,7 +981,7 @@ static void kinetis_txdone(FAR void *arg)
   /* Process TX completions */
 
   uint32_t mb_bit = 1 << RXMBCOUNT;
```
Comment (acassis) [inline arch/arm/src/kinetis/kinetis_flexcan.c:983]: same here

File: arch/arm/src/s32k1xx/s32k1xx_flexcan.c [hunk 4]
```diff
@@ -979,7 +982,7 @@ static void s32k1xx_txdone(FAR void *arg)
   /* Process TX completions */
 
   uint32_t mb_bit = 1 << RXMBCOUNT;
```
Comment (acassis) [inline arch/arm/src/s32k1xx/s32k1xx_flexcan.c:984]: I know you didn't touch this line in this commit, but please also move it to the top.

**Comment by PetervdPerk-NXP** [comment]
@acassis, you're right C89 style initialization is better. I've pushed a new commit.

**Comment by acassis** [comment]
> @acassis, you're right C89 style initialization is better. I've pushed a new commit.

Thank you for your modification Peter!

**Pull request merged**